src/Symfony/Component/PropertyInfo/PropertyInfoExtractor.php

```diff
@@ -39,17 +41,29 @@ class PropertyInfoExtractor implements PropertyInfoExtractorInterface
     private $accessExtractors;
 
     /**
+     * @var Cache
+     */
+    private $cache;
+
+    /**
+     * @var array
+     */
+    private $arrayCache = array();
+
+    /**
      * @param PropertyListExtractorInterface[]        $listExtractors
      * @param PropertyTypeExtractorInterface[]        $typeExtractors
      * @param PropertyDescriptionExtractorInterface[] $descriptionExtractors
      * @param PropertyAccessExtractorInterface[]      $accessExtractors
+     * @param Cache|null                              $cache
      */
-    public function __construct(array $listExtractors = array(), array $typeExtractors = array(),  array $descriptionExtractors = array(), array $accessExtractors = array())
+    public function __construct(array $listExtractors = array(), array $typeExtractors = array(),  array $descriptionExtractors = array(), array $accessExtractors = array(), Cache $cache = null)
     {
         $this->listExtractors = $listExtractors;
         $this->typeExtractors = $typeExtractors;
         $this->descriptionExtractors = $descriptionExtractors;
         $this->accessExtractors = $accessExtractors;
+        $this->cache = $cache;
```

Comment:
I was thinking the same thing initially, but finally I'm not sure that it's a good idea:

- Performance (and cache) is not an "optional" feature. It should be enable for all implementations in prod and IMO it's why it belongs to this class.
- In the context of the Standard Edition it adds a lot of complexity and an overhead: it requires to use the service decorator feature to switch the class depending of the environment (the cache must be enabled in prod, but not in dev)
- In the context of the component, it makes it harder to enable the cache (one class to instantiate)
- all other components using cache doesn't use composition for that (it is considered as a builtin feature)

When the PSR-6 will be ready, the argument can even be made mandatory (it's better to always use a cache) and add a `NullCache` or something like that can be added for the dev environment (like with the Logger).

What do you think @symfony/deciders?